### ISSUE TITLE
Train using validation set with weights

### ISSUE THREAD

**Issue by itamargolbary** [aws/sagemaker-xgboost-container#144]
https://github.com/aws/sagemaker-xgboost-container/blob/f57c886fa29fc826dbe401e1905ab03d657ff6cd/src/sagemaker_xgboost_container/algorithm_mode/train.py#L84

the validation set need to have weights in case of csv_weights=1.
see also this PR from xgboost: https://github.com/dmlc/xgboost/pull/2354
